docs/rules/prefer-find.md

```diff
@@ -0,0 +1,21 @@
+# Prefer `.find(…)` over first element of `.filter(…)` result
+
+[`Array#find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) breaks the loop as soon as it finds a match.
+
+This rule is fixable.
+
+## Fail
+
+```js
+const item = array.filter(x => x === '🦄')[0];
+```
```

Comment:
How about

```js
const [item] = array.filter(x => x === '🦄');
```

and

```js
const result = array.filter(x => x === '🦄');
const item = result[0];
```

?

Reply:
The first one:

```js
const [item = foo] = array.filter(x => x === '🦄');
```

-> 

```js
const item = array.find(x => x === '🦄') || foo;
```

?

Not safe, find result could be `0 | false | '' | null`

The second one seems more complicated.

Reply:
> Maybe just

👍